### ISSUE TITLE
Kartenladezeit Startseite manchmal sehr langsam

### ISSUE THREAD

**Issue by innosmith** [puzzle/decidim-zuerich#364]
Manchmal dauert das Laden der Startseite extrem lange, weil die Quartierkarte länger zum Laden braucht. Dieses Verhalten trifft ab und zu auf und äussert sich so, dass der Header und der Text links von der Quartierkarte geladen werden, der Rest der Seite aber noch weiss bleibt. Dieser Zustand kann 20 Sekunden oder länger dauern. Sobald die Karte geladen ist, wird auch der Rest der Startseite vollständig geladen.

Bitte prüfen: 
1) wie oft dieses Verhalten auftritt resp. in den Logs ersichtlich ist und damit ein Verbesserungsmassnahme gerechtfertigt ist
2) woran es liegen könnte und wie man dies verbessern kann

**Comment by carlobeltrame** [comment]
Das letzte Mal als ich das beobachtet habe lag es daran dass wir die Informationen, aus welchen die Karte zusammengesetzt wird, vom städtischen Server kommen, und dieser manchmal lange Reaktionszeiten hat. Es wäre aber vielleicht möglich, den Code der die Karte zusammensetzt so zu optimieren, dass nicht das Laden des Rests der Seite darauf warten muss. Müssten wir anschauen inwieweit das möglich ist. Die Karte selber wird dann aber trotzdem noch gleich lang zum Laden brauchen.

**Comment by innosmith** [comment]
Danke für die Rückmeldung @carlobeltrame . Konntest du aus dem LOG die Ladezeiten der Startseite nachvollziehen, dass dieses Problem öfter auftritt? Im Anschluss kontaktiere ich die Kollegen vom städtischen GIS-Server und frage nach, ob und was für Optionen sie sehen zur Optimierung.

**Comment by carlobeltrame** [comment]
Nein, das kann ich leider nicht. Die Karte wird im Browser zusammengesetzt, komplett unabhängig von unseren Servern, und das kritische Stück kommt ja vom stätdischen Server, was in unseren Logs nicht auftaucht.

**Comment by innosmith** [comment]
Alles klar. Ich hatte vermutet, man kann die Ladezeiten der Startseite als Ganzes aus den Logs lesen.

**Comment by carlobeltrame** [comment]
Das war vielleicht in den Neunzigern noch so, als noch praktisch nichts mit JavaScript und Animationen und clientseitig extern geladenen Daten gemacht wurde ;) Aber seither ist es nur schon schwierig, allgemein zu definieren wann Webseiten "fertig geladen" sind. Wenn dann könnte man das am ehesten in Analytics-Tools wie Matomo oder Sentry aufzeichnen, wenn man entsprechende Tracking-Features einrichtet. Aber die funktionieren dann auch wieder mit mehr JavaScript das auf der Seite ausgeführt werden muss.
